### PR TITLE
Try 2 xrootd caches before going back to the redirector

### PR DESCRIPTION
**Pull request by djw8605**

### CODE REVIEW

File: stashcp/__init__.py [hunk 1]
```diff
@@ -325,36 +325,52 @@ def download_xrootd(sourceFile, destination, debug, payload):
...
     if xrd_exit=='0': #worked first try
         logging.debug("Transfer success using %s", nearest_cache)
-        payload['tries'] = 1
+        status = "First Cache Success"
         payload['cache'] = cache
 
-    else: #pull from origin
-        logging.info("XrdCP from cache failed on %s, pulling from main redirector", nearest_cache)
+    if xrd_exit != '0': # pull from second nearest cache
+        cache = nearest_cache_list[1]
```
Comment (matyasselmeci) [inline stashcp/__init__.py:337]: This assumes that there are at least 2 caches in nearest_cache_list, which may not be true if using a custom caches.json file.

**Comment by matyasselmeci** [comment]
Why 2? Why not try all of them?

File: stashcp/__init__.py (hunk 1 above)
Comment (djw8605) [reply stashcp/__init__.py:337]: Good point.  Changed in f7df5a4

**Comment by djw8605** [comment]
It seems unnecessary to fail across the country.  There should some limit to the number of places to try.  Maybe we shouldn't go back to the origin in any case.  And try 3 caches?  I would rather try 3 caches (I like sets of 3) and not try the origin.

**Comment by matyasselmeci** [comment]
What does CVMFS do? IIRC, it tries multiple caches but never the origin?

**Comment by matyasselmeci** [comment]
If we don't want to replicate CVMFS's behavior, trying 3 caches but not the redirector is good enough.

**Comment by djw8605** [comment]
I updated the code to try 3 caches, in both HTTP and xrootd.

File: stashcp/__init__.py [hunk 2]
```diff
@@ -319,42 +319,34 @@ def download_xrootd(sourceFile, destination, debug, payload):
...
-
-        if xrd_exit=='0':
-            logging.info("Trunk Success")
-            status = 'Trunk Sucess'
-            tries=2
+    # Try 3 times to download from the 3 nearest caches
+    num_available_caches = len(nearest_cache_list)
+    tries = 0
+    xrd_exit = ""
+    for cache_idx in range(min(3, num_available_caches)): # try 3 caches, or how ever many caches are in the list
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply stashcp/__init__.py:327]: Also, doesn't this ignore the `--cache` flag from the command line? Relatedly, is `nearest_cache` now unused?

**Review by matyasselmeci — DISMISSED** [review]

File: stashcp/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply stashcp/__init__.py:327]: Wait, I'm wrong about the first point.

**Review by matyasselmeci — APPROVED** [review]
OK, it's good aside from the syntax error on the new `for` lines (not enough closing parens) and I'm not sure how it now handles the `--cache` command-line argument. Pre-approving.

**Pull request merged**